app/plugin.go

```diff
@@ -40,7 +40,12 @@ func (a *App) SyncPluginsActiveState() {
 
 			// If it's not enabled we need to deactivate it
 			if !pluginEnabled {
-				a.Plugins.Deactivate(pluginId)
+				deactivated := a.Plugins.Deactivate(pluginId)
```

Comment:
I moved the activate block below up to `SyncPluginsActiveState` since I needed the `updatedManifest`, and also realized that we might not be emitting these events consistently, e.g. when the config changes outside of the plugin API (say, `Enable` is set to false with the config watcher on).